lib/unit_tests_utils/postgresql_client.rb

```diff
@@ -0,0 +1,54 @@
+require 'pg'
+
+class PostgreSQLClient
+  attr_reader :host, :dbname, :user, :password, :sslmode
+
+  def initialize(args)
+    @host = args[:host]
+    @hostaddr = args[:hostaddr]
+    @dbname = args[:dbname] || 'area51'
+    @user = args[:user]
+    @password = args[:password]
+    @sslmode = args[:sslmode] || 'disable'
+  end
+
+  def self.create_from_manifest(manifest, host = nil)
+    host = manifest.hostname unless host
+    user = manifest.properties['postgresql-ha']['admin_credentials']['username']
+    password = manifest.properties['postgresql-ha']['admin_credentials']['password']
+    if manifest.properties['postgresql-ha']['ssl']['enable']
+      sslmode = "require"
+    end
+
+    self.new(host: host, user: user, password: password, sslmode: sslmode)
+  end
+
+  def ping(args = {})
+    args[:hostaddr] ||= args[:host] ? UnitTestsUtils::InternalDNS.resolve_domain_name(args[:host]).first : host
+    args[:dbname] ||= dbname
+    args[:user] ||= user
+    args[:password] ||= password
+    args[:sslmode] ||= sslmode
+
+    logger.debug("* Creating ping with args: *#{args}*")
+    return PG::Connection.ping(args)
+  end
+
+  def connect(args = {})
+    args[:hostaddr] ||= args[:host] ? UnitTestsUtils::InternalDNS.resolve_domain_name(args[:host]).first : host
```

Comment:
Should we use the args already created during initialization and skip this section supporting args override?
Are we using this for the current code?

My argument here would be, let's start as simple as possible, because the a9s PostgreSQL BOSH release has some complicated code that might not be necessary.

Reply:
The PGWebServiceClient is using the section supporting args override. However, I do not if it's ready necessary.
[see it](https://github.com/anynines/unit-tests-utils/blob/ds_561_a9s-pg_prepare_rspec/lib/unit_tests_utils/postgresql_web_service_client.rb#L188)


So, I will keep this session and clear the code not used. What do you thing ?